src/components/ThumbnailsPanel/ThumbnailsPanel.js

```diff
@@ -55,6 +71,75 @@ class ThumbnailsPanel extends React.PureComponent {
     });
   }
 
+  onDragEnd = () => {
+    const { currentPage } = this.props;
+    const { draggingOverPageIndex, isDraggingOverTopHalf } = this.state;
+    if (draggingOverPageIndex !== null) {
+      const targetPageNumber = isDraggingOverTopHalf ? draggingOverPageIndex + 1 : draggingOverPageIndex + 2;
+
+      if (currentPage < targetPageNumber) {
+        // moving page down so target page number will decrease
+        this.afterMovePageNumber = targetPageNumber - 1;
+      } else {
+        // moving page up
+        this.afterMovePageNumber = targetPageNumber;
+      }
+
+      core.movePages([currentPage], targetPageNumber);
+    }
+
+    this.setState({ draggingOverPageIndex: null });
+  }
+
+  onPageComplete = () => {
+    if (this.afterMovePageNumber) {
+      core.setCurrentPage(this.afterMovePageNumber);
+    }
+    this.afterMovePageNumber = null;
+  }
+
+  onDragOver = (e, index) => {
+    // prevent opening pdf dropped in
+    e.preventDefault();
+    const thumbnail = e.target.getBoundingClientRect();
+
+    if (e.pageY > (thumbnail.y + thumbnail.height / 2)) {
+      this.setState({ draggingOverPageIndex: index, isDraggingOverTopHalf: false });
+    } else {
+      this.setState({ draggingOverPageIndex: index, isDraggingOverTopHalf: true });
```

Comment:
Hmm the other way of writing this is
```javascript
this.setState({
  draggingOverPageIndex: index,
  isDraggingOverTopHalf: e.pageY > (thumbnail.y + thumbnail.height / 2)
})
```
This doesn't use ifelse, but I don't mind the current way.

Reply:
Thank you for pointing this out, I actually like your ternary statement more. I'll update it to that